### PR TITLE
Modify the metrics for CS smokedetector setup

### PR DESCRIPTION
**Pull request by tiffanny29631**
This change modifies metrics to align with Monarch definition

- includes rg_reconcile_duration_seconds in Monarch pipeline for future SLI setup

- clean up kcc_resource_count_total exclusion since the metric has been removed from Resource Group Controller

### CODE REVIEW

**Comment by tiffanny29631** [comment]
/hold

This change is blocked until a server side change is rolled out.

File: pkg/metrics/otel.go
Comment (haiyanmeng) [inline pkg/metrics/otel.go:150]: Did you test this locally to make sure that it does not cause any formatting issue (https://github.com/GoogleContainerTools/kpt-config-sync/commit/9ebabf3b4723a90b67c0ade9fa36b64fa380db45)?

Comment (tiffanny29631) [reply pkg/metrics/otel.go:150]: Yeah good point, I made sure there is no trailing spaces but will test to check the format

Comment (tiffanny29631) [reply pkg/metrics/otel.go:151]: Tested, format is without issue.

**Comment by tiffanny29631** [comment]
/unhold

Monarch change has rolled out, local test metric streaming with no issue.

**Review by haiyanmeng — APPROVED** [review]

**Comment by tiffanny29631** [comment]
@haiyanmeng Fixed the merge conflict, need lgtm again when you have time, thanks!

**Review by haiyanmeng — APPROVED** [review]

**Pull request merged**
